src/Pages/Articles.jsx

```diff
@@ -1,31 +1,65 @@
-import { useEffect, useState } from "react"
-import { ArticleCard } from "../Components/ArticleCard";
-import { getArticles } from "../utils/api";
+import { useState, useEffect } from "react"
+import { useParams } from "react-router-dom"
+import { ArticleCard } from '../Components/ArticleCard'
+import { getArticles } from '../utils/api'
 
-export const Articles = () => {
+ export const Topics = () => {
+    const [params, setParams] = useState({order: "asc"});
+    const [isLoading, setIsLoading] = useState(false);
     const [articles, setArticles] = useState([]);
-    const [isLoading, setIsLoading] = useState(false)
-    useEffect(() => {
-      setIsLoading(true)
-        getArticles()
-        .then((data) => {
-            setArticles(data.articles)
-            setIsLoading(false)
-        })
-    }, [])
-
-    if(isLoading) return <h4>Getting your articles ...</h4>
-
-return (
-  <section>
-    <ul>
-      {articles.map((article) => {
-        return (
-            <ArticleCard key={article.article_id} article={article} />
-        );
-      })}
-    </ul>
-  </section>
-);
-
-}
\ No newline at end of file
+    const [dropdown, setDropdown] = useState("")
+    const { topic_name } = useParams();
+  
+
+useEffect(() => {
+  setIsLoading(true)
+  getArticles({...params, topic: topic_name}).then((data) => {
+    setArticles(data.articles);
+    setIsLoading(false)
+  });
+}, [params, topic_name]);
+
+
+  
+  const handleOrder = () => {
+    setParams((currParams) => {
+      if (currParams.order === "asc") {
+        return {...currParams, order: "desc"}
+      } else {
+        return { ...currParams, order: "asc" };
+      }
+    })
+    };
+
+    const handleSubmit = () => {
+      if(dropdown !== '') {
+        setParams((currParams) => ({ ...currParams, sort_by: dropdown }));
+      }
+    }
+
+    const handleSort =(e)=>{
```

Comment:
I would argue that on selecting a new sort by option that should be the submit? You could move the logic from your handleSubmit and remove the submit button (you may have to play around with some refactoring) but I think from a user experience point of view you'd expect it to refresh when you made a new selection.

The code logic here is all in place - this is more about user experience!